lib/md051.js

```diff
@@ -119,12 +147,27 @@ module.exports = {
         for (const definition of definitions) {
           const { endColumn, startColumn } = definition;
           const text = unescapeStringTokenText(definition);
+          const textFragment = `#${encodeURIComponent(text.slice(1))}`
           if (
             (text.length > 1) &&
             text.startsWith("#") &&
-            !fragments.has(text) &&
-            !fragments.has(`#${encodeURIComponent(text.slice(1))}`)
+            !fragments.has(textFragment)
           ) {
+            if (textFragment.startsWith("#L")) {
+              const textFragmentLowercase = textFragment.toLowerCase()
+              if (fragments.has(textFragmentLowercase)) {
+                // eslint-disable-next-line no-continue
```

Comment:
I guess using `continue` in that context is fine? Nesting if blocks, makes the code difficult to read, and also generate a bigger diff, which is then harder to review.